src/Eval.hs

```diff
@@ -625,6 +625,8 @@ executeCommand ctx@(Context env _ _ _ _ _ _ _) xobj =
         executeCommand newCtx (withBuildAndRun (XObj (Lst []) (Just dummyInfo) Nothing))
       Left (HasStaticCall _ _) ->
         callFromRepl newCtx xobj
+      Right a@(XObj (Arr _) _ _) ->
+        executeCommand newCtx (XObj (Lst [XObj (Sym (SymPath [] "str") Symbol) (Just dummyInfo) Nothing, XObj (Lst [XObj (Sym (SymPath [] "quote") Symbol) (Just dummyInfo) Nothing, a]) (Just dummyInfo) Nothing]) (Just dummyInfo) Nothing)
```

Comment:
What does this impressive line of code do? :D

Reply:
Ah yes, that requires cleaning up. Basically what it says is that if we return an array from the top-level evaluator, we call it again with `(str '<array)`. This should be safe, since arrays are "just" data, I think?! What it enables is that `str` then decides whether this is a dynamic type it knows how to stringify or not!

Reply:
I see, thanks for the explanation! The array can contain function calls, of course – but I gather that that should be fine?

Reply:
If the array is returned, those shouldn’t be  evaluated anymore anyway. Quoting the array should lead to the same effect I think.